src/fireedge/src/modules/utils/merge.js

```diff
@@ -0,0 +1,156 @@
+ * @param {Array} source - Source
+ * @param {MergeOptions} options - Merge options
+ * @returns {Array} Two arrays merged
+ */
+export const defaultArrayMerge = (target, source, options) => {
+  const destination = target.slice()
+  source.forEach(function (e, i) {
+    if (typeof destination[i] === 'undefined') {
+      destination[i] = cloneIfNecessary(e, options)
+    } else if (isMergeableObject(e)) {
+      destination[i] = deepmerge(target[i], e, options)
+    } else if (target.indexOf(e) === -1) {
+      destination.push(cloneIfNecessary(e, options))
+    }
+  })
+
+  return destination
+}
+
+/**
+ * @param {object} target - Target
+ * @param {object} source - Source
+ * @param {MergeOptions} options - Merge options
+ * @returns {object} Two object merged
+ */
+export const mergeObject = (target, source, options) => {
+  const destination = {}
+  if (isMergeableObject(target)) {
+    Object.keys(target).forEach(function (key) {
+      destination[key] = cloneIfNecessary(target[key], options)
+    })
+  }
+  Object.keys(source).forEach(function (key) {
+    if (!isMergeableObject(source[key]) || !target[key]) {
+      destination[key] = cloneIfNecessary(source[key], options)
+    } else {
+      destination[key] = deepmerge(target[key], source[key], options)
+    }
+  })
+
+  return destination
+}
+
+/**
+ * Merges the enumerable properties of two or more objects deeply.
+ *
+ * @param {object} target - Target
+ * @param {object} source - Source
+ * @param {MergeOptions} options - Merge options
+ * @returns {object} Two object merged
+ */
+export const deepmerge = (target, source, options = {}) => {
+  const array = Array.isArray(source)
+  const { arrayMerge = defaultArrayMerge } = options
+
+  if (array) {
+    return Array.isArray(target)
+      ? arrayMerge(target, source, options)
+      : cloneIfNecessary(source, options)
+  } else {
+    return mergeObject(target, source, options)
+  }
+}
+
+deepmerge.all = function deepmergeAll(array, options) {
+  if (!Array.isArray(array) || array.length < 2) {
+    throw new Error(
+      'first argument should be an array with at least two elements'
+    )
+  }
+
+  // we are sure there are at least 2 values, so it is safe to have no initial value
+  return array.reduce(function (prev, next) {
+    return deepmerge(prev, next, options)
+  })
+}
+
+/**
+ * Flattens specified keys in an object by one level without mutating the original object.
+ *
+ * @param {object} obj - The object to flatten.
+ * @param {Array} keysToFlatten - The keys whose nested objects should be flattened.
+ * @returns {object} The new flattened object.
+ */
+export const flattenObjectByKeys = (obj, keysToFlatten) => {
+  let result = { ...obj }
+
+  keysToFlatten.forEach((key) => {
+    if (
+      key in result &&
+      typeof result[key] === 'object' &&
+      result[key] !== null
+    ) {
+      result = {
+        ...result,
+        ...result[key],
+      }
+      delete result[key]
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

We detected that you are using `delete` to remove items from an array. If we use `delete`, it will delete the object property, but will not reindex the array or update its length. It leaves `undefined` holes in the array. Therefore, we recommend you to use `splice` to remove items from an array.

[Learn more](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/delete)